samples/pmic/npm6001/src/main.c

```diff
@@ -0,0 +1,530 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+
+#include <dk_buttons_and_leds.h>
+
+#include <lib_npm6001.h>
+
+LOG_MODULE_REGISTER(main, CONFIG_MAIN_LOG_LEVEL);
+
+#define LED_INIT_OK DK_LED1
+#define LED_BUCK3_ON DK_LED2
+#define LED_GPIO_ON DK_LED4
+
+#define BUCK0_DEFAULT_VOLTAGE 3300
+#define BUCK1_DEFAULT_VOLTAGE 1200
+#define BUCK2_DEFAULT_VOLTAGE 1400
+#define BUCK3_DEFAULT_VOLTAGE 500
+#define LDO0_DEFAULT_VOLTAGE 2700
+
+K_SEM_DEFINE(n_int_sem, 0, 1);
+
+static const struct device *i2c_dev = DEVICE_DT_GET_ONE(nordic_nrf_twim);
+
+static const struct gpio_dt_spec n_int = GPIO_DT_SPEC_GET(DT_ALIAS(nint), gpios);
+static struct gpio_callback n_int_callback;
+
+static uint16_t buck3_voltage;
+static bool buck3_enabled;
+static bool npm6001_gpio_set;
+
+static int lib_npm6001_platform_init(void)
+{
+	if (i2c_dev == NULL) {
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int lib_npm6001_twi_read(uint8_t *buf, uint8_t len, uint8_t reg_addr)
+{
+	int err;
+
+	if (IS_ENABLED(CONFIG_TEST)) {
+		memset(buf, 0, len);
+		return len;
+	}
```

Comment:
?

Reply:
I wanted a simple I2C emulation without creating an emulated driver and compatible board for the zephyr i2c emulator (https://docs.zephyrproject.org/3.1.0/hardware/emulator/index.html)